be/src/vec/functions/function_regexp.cpp

```diff
@@ -65,6 +66,281 @@ struct FourParamTypes {
+        }
+    }
+
+    // 全常量（pattern & start_str 都为常量）的执行路径
+    static void execute_impl_const_args(FunctionContext* context, ColumnPtr argument_columns[],
+                                        size_t argument_size, size_t input_rows_count,
+                                        ColumnInt64::Container& result_data, NullMap& null_map) {
+        const auto* str_col = check_and_get_column<ColumnString>(argument_columns[0].get());
+        const auto* pattern_col = check_and_get_column<ColumnString>(argument_columns[1].get());
+        const ColumnString* start_str_col = nullptr;
+        Int64 start_pos_const = 1;
+
+        if (argument_size == 3) {
+            start_str_col = check_and_get_column<ColumnString>(argument_columns[2].get());
+            auto start_data = start_str_col->get_data_at(0);
+            std::string start_str(start_data.data, start_data.size);
+            try {
+                start_pos_const = std::stoll(start_str);
+            } catch (...) {
+                for (size_t i = 0; i < input_rows_count; ++i) {
+                    result_data[i] = -1;
+                }
+                return;
+            }
+            if (start_pos_const < 1) {
+                for (size_t i = 0; i < input_rows_count; ++i) {
+                    result_data[i] = -1;
+                }
+                return;
+            }
+        }
+
+        re2::RE2* re = reinterpret_cast<re2::RE2*>(
+                context->get_function_state(FunctionContext::THREAD_LOCAL));
+        if (!re) {
+            for (size_t i = 0; i < input_rows_count; ++i) {
+                result_data[i] = -1;
+            }
+            return;
+        }
+
+        for (size_t i = 0; i < input_rows_count; ++i) {
+            if (null_map[i]) {
+                result_data[i] = -1;
+                continue;
+            }
+            auto str_data = str_col->get_data_at(i);
+            re2::StringPiece str_sp(str_data.data, str_data.size);
+
+            size_t search_start = static_cast<size_t>(start_pos_const - 1);
+            if (search_start >= str_data.size) {
+                result_data[i] = -1;
+                continue;
+            }
+
+            re2::StringPiece match;
+            bool success =
+                    re->Match(str_sp, search_start, str_data.size, re2::RE2::UNANCHORED, &match, 1);
+            if (!success) {
+                result_data[i] = -1;
+                continue;
+            }
+
+            size_t match_pos = match.data() - str_data.data + 1;
+            result_data[i] = static_cast<Int64>(match_pos);
+        }
+    }
+};
+
+// -----------------------------------------------------------------------------
+// FunctionRegexpPosition：包装 IFunction 接口
+// -----------------------------------------------------------------------------
+
+// >>> Added: 新增 FunctionRegexpPosition 类，用于在 ClickHouse 中注册 regexp_position
+class FunctionRegexpPosition : public IFunction {
```

Comment:
this class has only one type implementation: `RegexpPositionImpl`. no need to split them